### PR TITLE
Feat: Date range input

### PR DESCRIPTION
**Pull request by vorillaz**
### Background

This is the tailored made implementation of a Date range input component. It allows picking a date range with time and even allows presets with preselected date ranges.

### Screenshots
Screenshot 2020-08-31 at 8 36 14 PM
demo

### Designs 
https://app.abstract.com/projects/ab14d6c7-511b-4577-8de9-25ca9cd2918b/branches/b748d896-1a04-4c1c-8137-0471aaa8fd54/commits/c172b4b5b268ca521b77f440bd609ef4713fa45f/files/d379a6e4-1ffd-4b9f-adc8-22e5463c4961/layers/C80BD105-F98C-4664-BFF9-F73E8C7FD84A

### Testing

- Manually and visually
- Unit tests.

### CODE REVIEW

File: src/components/DateInput/DateInput.test.tsx [hunk 1]
```diff
@@ -38,12 +38,12 @@ describe('DateInput', () => {
     await fireEvent.click(input);
 
     expect(await findByLabelText('Su Nov 01 2020')).toBeInTheDocument();
-    const prev = await findByLabelText('Go to previous page');
+    const prev = await findByLabelText('Go to previous month');
```
Comment (vorillaz) [inline src/components/DateInput/DateInput.test.tsx:41]: Better labels.

File: src/components/DateInput/DateInput.tsx [hunk 2]
```diff
@@ -110,62 +110,70 @@ const DateInput: React.FC<DateInputProps & Omit<TextInputProps, 'value' | 'onCha
...
-        <Box onClick={onExpand} cursor="pointer">
-          <TextInput
-            {...rest}
-            value={currentDate ? dayjs(currentDate).format(format) : ''}
-            autoComplete="off"
-            icon="calendar"
-            aria-autocomplete="none"
-            tabIndex={-1}
-            readOnly
+    <Box position="relative">
```
Comment (vorillaz) [inline src/components/DateInput/DateInput.tsx:113]: Reorder the DateInput bits and pieces for better composition.

File: src/components/DateInput/Day.tsx [hunk 3]
```diff
@@ -1,35 +1,105 @@
...
-    _selected={{ backgroundColor: 'blue-400' }}
-    _hover={{ backgroundColor: 'blue-300' }}
+    position="relative"
     {...props}
   />
 );
 
-const MemoCell = React.memo(Cell);
+const OuterCell = styled(PseudoOuterCell)`
+  [aria-relevant]:hover {
```
Comment (vorillaz) [inline src/components/DateInput/Day.tsx:29]: I used aria attributes in order to create the visual effect of the range.

File: src/components/DateInput/Day.tsx [hunk 4]
```diff
@@ -1,35 +1,105 @@
...
+
+  &[aria-busy='true'] + [aria-selected='true'],
+  &[aria-selected='true'] + [aria-busy='true'],
+  &[aria-selected='true'] + &[aria-selected='true'] {
+    [aria-placeholder] {
+      background-color: ${({ theme }) => addOpacity(theme.colors['navyblue-100'], 0.2)};
+    }
+  }
+
+  &[aria-selected='true'] + &[aria-selected='true'] [aria-placeholder] {
```
Comment (vorillaz) [inline src/components/DateInput/Day.tsx:48]: Lots of selectors in order to show the placeholder as a range along with border-radius for each row.

**Comment by 3nvi** [comment]
Designs: https://app.abstract.com/projects/ab14d6c7-511b-4577-8de9-25ca9cd2918b/branches/b748d896-1a04-4c1c-8137-0471aaa8fd54/commits/c172b4b5b268ca521b77f440bd609ef4713fa45f/files/d379a6e4-1ffd-4b9f-adc8-22e5463c4961/layers/C80BD105-F98C-4664-BFF9-F73E8C7FD84A

File: src/components/DateInput/Day.tsx [hunk 5]
```diff
@@ -1,35 +1,105 @@
...
+  &[aria-selected='true'] + &[aria-busy='true'] [aria-placeholder] {
+    left: -50%;
+  }
+
+  &[aria-selected='true'] [aria-relevant] {
+    background-color: ${({ theme }) => theme.colors['blue-400']};
+  }
+  &[aria-busy='true'] [aria-relevant] {
+    border-radius: 0;
+    background-color: ${({ theme }) => addOpacity(theme.colors['navyblue-100'], 0.2)};
```
Comment (vorillaz) [inline src/components/DateInput/Day.tsx:64]: Added color opacity through the theme.

File: src/components/DateInput/Day.tsx [hunk 6]
```diff
@@ -1,35 +1,105 @@
...
+  }
+  &[aria-disabled='true'] + &[aria-busy='true'] {
+    [aria-relevant] {
+      border-radius: 4px 0 0 4px;
+    }
+  }
+  &[aria-busy='true'] {
+    &:last-of-type [aria-relevant] {
+      border-radius: 0 4px 4px 0;
```
Comment (vorillaz) [inline src/components/DateInput/Day.tsx:73]: 😬  Fix radiuses per row and selected date.

File: src/components/DateInput/Day.tsx [hunk 7]
```diff
@@ -1,35 +1,105 @@
...
+  month,
+  year,
+  daySelected,
+  isLastRow,
+  dayRangeSelected,
+  onDaySelect = noop,
+}) => {
   if (!day) {
-    return <MemoCell aria-disabled="true" />;
+    return isLastRow ? null : <MemoOuterCell aria-disabled="true" />;
```
Comment (vorillaz) [inline src/components/DateInput/Day.tsx:102]: We need to ensure that there are now extra disabled cells as we want the pseudo CSS selectors take action, since this is a table the cells from the previous rows are holding everything in place. 
TL;DR This is used in order to make border-radius work

File: src/components/DateRangeInput/DoubleTextInput.tsx [hunk 8]
```diff
@@ -0,0 +1,132 @@
+import React, { ChangeEventHandler } from 'react';
```
Comment (vorillaz) [inline src/components/DateRangeInput/DoubleTextInput.tsx:1]: A grouped input component. Labels, names, placeholders and ids may get generated.

File: src/components/DateRangeInput/Presets.tsx [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+    _hover={{ color: 'gray-50' }}
+    aria-selected={selected}
+    onClick={onSelect}
+    fontSize="medium"
+    {...rest}
+  />
+);
+
+const now = dayjs();
+const lastDay = now.subtract(1, 'day');
```
Comment (vorillaz) [inline src/components/DateRangeInput/Presets.tsx:33]: Presets as per design needs.

File: src/components/DateRangeInput/DateRangeInput.tsx [hunk 10]
```diff
@@ -0,0 +1,316 @@
...
+      return setCurrentRange([currentDateRange[0], dateChanged.toDate()]);
+    },
+    [setCurrentRange, currentDateRange]
+  );
+
+  const isDisabled = useCallback(() => !currentDateRange || currentDateRange?.length < 2, [
+    currentDateRange,
+  ]);
+
+  const onPresetSelect = useCallback(
```
Comment (3nvi) [inline src/components/DateRangeInput/DateRangeInput.tsx:176]: Is there a chance we can "move" the month  screen depending on thhe selection? For example, if we are in the

"September - October" view and I  select "last 2 weeks" is it easy for the view to go to "August - September" with the days highlighted?

If it's not trivial  we can mark it as  an enhancement, but it helps people not get lost ( i.e. I was browsing months so I reached December 2020, I clicked "last 24h" and I didn't see anything change so I thought I found a bug. If the calendar month automatically went to today's month, I'd  be able to understand that the option has been correctly selected)

File: src/components/DateInput/Day.tsx (hunk 7 above)
Comment (vorillaz) [reply src/components/DateInput/Day.tsx:102]: @3nvi If a range is selected we need to highlight the cell between the given range. The first and the last cells of each row should get a tiny border-radius as seen here:
Screenshot 2020-09-01 at 4 57 24 PM
 
This styling is applied using `:last-of-type` and `:first-of-type` pseudo-selectors, but in order to use the `:one-of-type` pseudo selector when there is just one cell in the last row we need to discard any disabled cells.

Comment (3nvi) [reply src/components/DateInput/Day.tsx:102]: > but in order to use the :one-of-type pseudo selector when there is just one cell in the last row we need to discard any disabled cells.

I think I understand it. Right now each month has some disabled dates (that belong to the next month) which are shown as disabled. In order for the CSS rule to apply you want to "hide" those days completely right?

Where  my  thought process fails  though, is that I see that you never  show any dates as disabled. When is this the case?

File: src/components/DateInput/Day.tsx [hunk 11]
```diff
@@ -1,35 +1,105 @@
...
+    &:first-of-type [aria-relevant] {
+      border-radius: 4px 0 0 4px;
+    }
+    &:only-of-type [aria-relevant] {
+      border-radius: 4px;
+    }
+  }
+`;
+
+const MemoOuterCell = React.memo(OuterCell);
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateInput/Day.tsx]: Yes we did 😆

File: src/components/DateRangeInput/Presets.tsx (hunk 9 above)
Comment (vorillaz) [reply src/components/DateRangeInput/Presets.tsx]: You are right, thus I created a helper method that generates the data on the call site.

File: src/components/DateRangeInput/Presets.tsx [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+const lastWeek = now.subtract(1, 'week');
+const lastMonth = now.subtract(1, 'month');
+const lastThreeMonths = now.subtract(3, 'month');
+const lastSixMonths = now.subtract(6, 'month');
+
+const options = [
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateRangeInput/Presets.tsx:34]: @3nvi Since we need to ensure that each range is visible I am dynamically changing the current month in order to highlight the preset selected.

File: src/components/DateRangeInput/Presets.tsx [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+  onSelect: (dates: [Dayjs, Dayjs]) => void;
+}
+
+interface ListItemProps {
+  selected?: boolean;
+  onSelect: (evt: React.KeyboardEvent) => void;
+}
+
+const ListItem: React.FC<ListItemProps> = ({ selected, onSelect, ...rest }) => (
+  <Box
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateRangeInput/Presets.tsx]: Thanks for the suggestion, fixed this one!

File: src/components/DateRangeInput/Presets.tsx [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+
+    if (start.isSame(lastSixMonths, 'date')) {
+      return 'last_six_months';
+    }
+    return 'custom';
+  }, [currentDateRange]);
+  return (
+    <Box borderRight="1px solid" borderColor="navyblue-300">
+      <Box p={6} width="140px">
+        <Box as="ul">
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateRangeInput/Presets.tsx]: Fixed as well :)

File: src/components/DateRangeInput/Presets.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (3nvi) [reply src/components/DateRangeInput/Presets.tsx:34]: I don't think this is related to what I mention. The "last 24h" should occur from the time   I clicked and **not** from the time the I loaded the tab that  serves the website. 

Am I misunderstanding something in your response?

File: src/components/DateInput/Day.tsx [hunk 15]
```diff
@@ -41,23 +111,41 @@ const Day: React.FC<DayProps> = ({ day, month, year, daySelected, onDaySelect =
   );
 
   const isSelected = React.useCallback(() => {
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateInput/Day.tsx]: Done!

File: src/components/DateInput/Day.tsx [hunk 16]
```diff
@@ -1,35 +1,105 @@
...
+    }
+    &:only-of-type [aria-relevant] {
+      border-radius: 4px;
+    }
+  }
+`;
+
+const MemoOuterCell = React.memo(OuterCell);
+
+const PseudoCell: React.FC<BoxProps> = props => (
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateInput/Day.tsx]: Used fixed width and height in order to create circles around the highlighted items.

File: src/components/DateRangeInput/DateRangeInput.tsx (hunk 10 above)
Comment (vorillaz) [reply src/components/DateRangeInput/DateRangeInput.tsx:175]: It's done, selected months are always in view as presets are changing.

File: src/components/DateInput/Day.tsx [hunk 17]
```diff
@@ -41,23 +111,41 @@ const Day: React.FC<DayProps> = ({ day, month, year, daySelected, onDaySelect =
...
+    <MemoOuterCell
       aria-label={date.format('dd MMM DD YYYY')}
       onClick={onDaySelectClick}
+      role="gridcell"
+      aria-busy={isWithinRange()}
+      aria-selected={isSelected()}
     >
-      {day}
-    </MemoCell>
+      <Box aria-placeholder="true" tabIndex={-1} pointerEvents="none" />
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateInput/Day.tsx:85]: This is the trailing or the leading highlighted background for the selected dates within the range.
The date gets highlighted as a circle and we shall move the "placeholder accordingly in order to create a stacked  effect.
Screenshot 2020-09-02 at 2 09 09 PM

File: src/components/DateRangeInput/DoubleTextInput.tsx [hunk 18]
```diff
@@ -0,0 +1,132 @@
...
+import Box, { NativeAttributes } from '../Box';
+import { slugify } from '../../utils/helpers';
+import Icon, { IconProps } from '../Icon';
+import Flex from '../Flex';
+import { InputControl, InputElement, InputLabel } from '../utils/Input';
+import { noop } from '../../utils/helpers';
+
+export type DoubleTextInputProps = NativeAttributes<'input'> & {
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateRangeInput/DoubleTextInput.tsx]: Yeap, you are right! Thanks on this one, fixed.

File: src/components/DateRangeInput/Presets.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateRangeInput/Presets.tsx:34]: I fixed this one as it can be seen here
https://github.com/panther-labs/pounce/blob/d4ed50ec62262da52ba20666aecef75dca244d0a/src/components/DateRangeInput/Presets.tsx#L42

File: src/components/DateRangeInput/DateRangeInput.mdx [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+  );
+};
+
+<DateRangeExample />;
+```
+
+```typescript jsx
+const format = 'MM/DD/YYYY hh:mm A';
+const DateRangeExample = () => {
+  const [dates, setDates] = React.useState([new Date(), new Date(Date.now() + 12096e5)]);
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateRangeInput/DateRangeInput.mdx:72]: This error is comping from the `styleguidist`, dates are manipulated explicitly by dayjs.
I updated the actual examples using an empty array as an example which makes sense IMHO
https://github.com/panther-labs/pounce/blob/bc0bc336953891d07f163d8548b99a511f0dd058/src/components/DateRangeInput/DateRangeInput.mdx#L10

File: src/components/DateInput/Month.tsx [hunk 20]
```diff
@@ -70,26 +84,30 @@ const Month: React.FC<MonthProps> = ({ year, month, daySelected, onDaySelect = n
...
+            <Box display="table-row" key={`${year}-${month}-${monthIndex}-week`}>
+              {week.map((day, dayIndex) => (
+                <Day
+                  onDaySelect={onDaySelect}
+                  dayRangeSelected={dayRangeSelected}
+                  daySelected={daySelected}
+                  year={year}
+                  month={month}
+                  day={day}
+                  isLastRow={weeks.length === monthIndex + 1}
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateInput/Month.tsx:93]: Last week row is invoked in order to limit cells with specific styles, disabled cells working as placeholders are affecting the `:last-of-type` selectors in the last row of the month view.

File: src/components/DateInput/Month.tsx [hunk 21]
```diff
@@ -70,26 +84,30 @@ const Month: React.FC<MonthProps> = ({ year, month, daySelected, onDaySelect = n
...
-      </Box>
+      <Table>
+        <Box display="table-row-group">
+          {weeks.map((week, monthIndex) => (
+            // eslint-disable-next-line
+            <Box display="table-row" key={`${year}-${month}-${monthIndex}-week`}>
+              {week.map((day, dayIndex) => (
+                <Day
+                  onDaySelect={onDaySelect}
+                  dayRangeSelected={dayRangeSelected}
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateInput/Month.tsx:88]: Yeap, the `Day` component is aware of the range in order to pick the appropriate styles.
We should decide if a Day is an in-between range, selected or disabled.

File: src/components/DateInput/Day.tsx (hunk 17 above)
[the remark this answers is not in the document]
Comment (3nvi) [reply src/components/DateInput/Day.tsx:85]: hmm  I'm thinking of ways to make this more evident, since an  `aria-placeholder` on a `Box` is not intuitive that's **only** creating a "stylistic" change.

Can we add  a pseudoelement  on the `Day` component to do that? We could use  `_before` or `_after`

File: src/components/DateRangeInput/DateRangeInput.mdx (hunk 19 above)
[the remark this answers is not in the document]
Comment (3nvi) [reply src/components/DateRangeInput/DateRangeInput.mdx:72]: Cool, if empty array is a valid value, I'm perfectly ok 😃

File: src/components/DateInput/Month.tsx (hunk 21 above)
[the remark this answers is not in the document]
Comment (3nvi) [reply src/components/DateInput/Month.tsx:88]: would it make sense for the `Day` not to be aware  of that, but to pass booleans like `highlighted`, `selected` & `disabled`?

My sole reasoning is that we can "dumb-down" this component and make it a bit more agnostic of the entire integration. Thoughts?

File: src/components/DateInput/Day.tsx [hunk 22]
```diff
@@ -1,35 +1,105 @@
...
+    [aria-placeholder] {
+      background-color: ${({ theme }) => addOpacity(theme.colors['navyblue-100'], 0.2)};
+    }
+  }
+
+  &[aria-selected='true'] + &[aria-selected='true'] [aria-placeholder] {
+    width: 100%;
+    left: -50%;
+  }
+  &[aria-busy='true'] + &[aria-selected='true'] [aria-placeholder] {
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateInput/Day.tsx]: These are relevant selectors in order to achieve the effect of the highlighted range. The CSS is a bit complicated and weird as we need to accommodate different scenarios like adding border-radius for days at the start or the end of a week row, adding a trailing or leading highlighting element for each date selected and so on.

File: src/components/DateRangeInput/DateRangeInput.test.tsx [hunk 23]
```diff
@@ -0,0 +1,121 @@
...
+    <DateRangeInput {...props} withTime onChange={mock} />
+  );
+  const input = await findByLabelText('From date');
+
+  // Open the date input components
+  await fireEvent.click(input);
+  expect(container).toMatchSnapshot();
+});
+
+it('allows passing a custom format', async () => {
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateRangeInput/DateRangeInput.test.tsx:69]: The snapshot is validating the actual format :)

https://github.com/panther-labs/pounce/blob/cc5cf74d08ae8841ffae8d721d797a70196f8b02/src/components/DateRangeInput/__snapshots__/DateRangeInput.test.tsx.snap#L176-L186

File: src/components/DateRangeInput/DateRangeInput.test.tsx [hunk 24]
```diff
@@ -0,0 +1,121 @@
...
+  const mock = jest.fn();
+  const { container } = await renderWithTheme(
+    <DateRangeInput {...props} format="M MMM YYYY D dd" onChange={mock} />
+  );
+  expect(container).toMatchSnapshot();
+});
+
+it('allows selecting a preset', async () => {
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateRangeInput/DateRangeInput.test.tsx:77]: You were right since the tests are time-specific I updated the actual assertions as:
https://github.com/panther-labs/pounce/blob/bbcb3c0a3be4d5d43ce3f56cc1de452a0520fe90/src/components/DateRangeInput/DateRangeInput.test.tsx#L69-L104

File: src/components/DateRangeInput/DoubleTextInput.test.tsx [hunk 25]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateRangeInput/DoubleTextInput.test.tsx:1]: And it's done
https://github.com/panther-labs/pounce/blob/fc805aeb84d1bb97ff814d27a5f1e91844e651d6/src/components/DateRangeInput/DoubleTextInput.test.tsx#L37-L68

File: src/components/DateInput/Day.tsx (hunk 7 above)
Comment (vorillaz) [reply src/components/DateInput/Day.tsx]: I tried using `<Box _after />` but placeholders are affected by their siblings, thus we can't effectively style them.
Furthermore, styling gets even more bloated.

File: src/components/DateRangeInput/DateRangeInput.test.tsx (hunk 23 above)
[the remark this answers is not in the document]
Comment (3nvi) [reply src/components/DateRangeInput/DateRangeInput.test.tsx:69]: yeah but how do we validate that the value is indeed changing **correctly** when I change the format.

I'd expect that we need to do `expect(input.value).toEqual(dayjs(date, format))`

File: src/components/DateRangeInput/DateRangeInput.test.tsx [hunk 26]
```diff
@@ -0,0 +1,121 @@
...
+
+  await fireEvent.click(cellStart);
+  await fireEvent.click(cellEnd);
+  await fireEvent.click(submitBtn);
+
+  expect(mock).toHaveBeenCalled();
+  expect(container).toMatchSnapshot();
+});
+
+it('allows selecting a date range', async () => {
```
[the remark this answers is not in the document]
Comment (3nvi) [reply src/components/DateRangeInput/DateRangeInput.test.tsx:106]: @vorillaz ping

[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateRangeInput/DateRangeInput.test.tsx:106]: Fixed as the preset testing validates the actual selection.

File: src/components/DateInput/TimePicker.tsx [hunk 27]
```diff
@@ -44,35 +43,33 @@ const TimePicker: React.FC<TimePickerProps> = ({ date, onTimeUpdate }) => {
   );
 
   return (
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateInput/TimePicker.tsx:50]: This is kinda weird imho, using an outer `<Flex />` allows us to format the actual boxes. This is essential as we want to use different layouts for the `DateInput` and the `DateRangeInput` components accordingly.

File: src/components/DateRangeInput/DateRangeInput.test.tsx [hunk 28]
```diff
@@ -0,0 +1,121 @@
...
+  expect(await findByLabelText('Th Oct 01 2020')).toBeInTheDocument();
+  const next = await findByLabelText('Go to next month');
+
+  // Go to December
+  await next.click(input);
+  await next.click(input);
+  expect(await findByLabelText('We Dec 02 2020')).toBeInTheDocument();
+});
+
+it('allows selecting a date with time options', async () => {
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateRangeInput/DateRangeInput.test.tsx]: The specs have been updated with more scenarios and proper naming.

File: src/components/DateRangeInput/DateRangeInput.test.tsx (hunk 23 above)
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateRangeInput/DateRangeInput.test.tsx:107]: Extended the testing scenarios as well.

File: src/components/DateRangeInput/DateRangeInput.tsx [hunk 29]
```diff
@@ -0,0 +1,316 @@
...
+            {withTime && (
+              <Flex
+                align="center"
+                justify="center"
+                borderTop="1px solid"
+                borderColor="navyblue-300"
+                p={4}
+                spacing={8}
+              >
+                <Flex align="center" justify="center" spacing={3}>
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateRangeInput/DateRangeInput.tsx:284]: The layout is a bit different per call site of the `TimePicker`. It's a bit different for the `DateInput` and the `DateRangeInput` consumers, I don't this it's a big deal leave them fluid and use a `Flex` component on the outer scope.

File: src/components/DateRangeInput/Presets.tsx [hunk 30]
```diff
@@ -0,0 +1,139 @@
...
+      return 'last_three_months';
+    }
+
+    if (start.isSame(lastSixMonths, 'date')) {
+      return 'last_six_months';
+    }
+    return 'custom';
+  }, [currentDateRange]);
+  return (
+    <Box borderRight="1px solid" borderColor="navyblue-300">
```
[the remark this answers is not in the document]
Comment (vorillaz) [reply src/components/DateRangeInput/Presets.tsx:137]: Apparently no. The first one is used in order to create the column effect.
The width of the second one is used as flex shrinking was breaking the layout. And finally list elements (`ol`, `ul`) are not playing well with spacing and fixed width.

File: src/components/DateRangeInput/DateRangeInput.test.tsx (hunk 23 above)
[the remark this answers is not in the document]
Comment (3nvi) [reply src/components/DateRangeInput/DateRangeInput.test.tsx:107]: 👍

File: src/components/DateInput/Month.tsx (hunk 21 above)
[the remark this answers is not in the document]
Comment (3nvi) [reply src/components/DateInput/Month.tsx:88]: It's not something that should hinder the release of this, but definitely something we should consider. Let's write it down as an enhancement

File: src/components/DateInput/Day.tsx [hunk 31]
```diff
@@ -1,35 +1,40 @@
...
+  daySelected,
+  isLastRow,
+  dayRangeSelected,
+  onDaySelect = noop,
+}) => {
   if (!day) {
-    return <MemoCell aria-disabled="true" />;
+    return isLastRow ? null : (
+      <Cell
+        display="table-cell"
```
Comment (3nvi) [inline src/components/DateInput/Day.tsx:30]: `cursor`,  `textAlign` and `verticalAlign` are un-needed here

File: src/components/DateInput/Day.tsx [hunk 32]
```diff
@@ -41,23 +111,41 @@ const Day: React.FC<DayProps> = ({ day, month, year, daySelected, onDaySelect =
...
-  }, [date, daySelected]);
+    if (daySelected) {
+      return date.isSame(daySelected, 'date');
+    }
+    // @ts-ignore
+    const [start, end] = dayRangeSelected;
+    return (start && date.isSame(start, 'date')) || (end && date.isSame(end, 'date'));
+  }, [date, daySelected, dayRangeSelected]);
+
+  const isWithinRange = React.useCallback(() => {
```
[the remark this answers is not in the document]
Comment (3nvi) [reply src/components/DateInput/Day.tsx:64]: This hasn't been addressed 😅

**Review by 3nvi — APPROVED** [review]
Thanks for addressing the review comments @vorillaz 

I'm approving although I still feel strongly about the picker being able to close when you click outside. 

Οther than that:

- We should explore how we can remove the extra `<div />` that just serve the potential border radius (currently present in **all** cells)
- Try and making some components dumber
